### PR TITLE
Initial support for XDG output protocol

### PR DESCRIPTION
**Pull request by wmww**
Disabled by default. To test, add `"zxdg_output_v1"` to recommended extensions in [examples/mir_demo_server/server_example.cpp](examples/mir_demo_server/server_example.cpp)

EDIT: or just add it to `--wayland-extensions`

### CODE REVIEW

File: src/wayland/symbols.map [hunk 1]
```diff
@@ -205,3 +205,21 @@ global:
   };
   local: *;
 };
+
+MIRWAYLAND_1.3 {
```
Comment (AlanGriffiths) [inline src/wayland/symbols.map]: We've not released 1.2 yet, so you can add to that.

**Comment by AlanGriffiths** [comment]
> To test, add `"zxdg_output_v1"` to recommended extensions in [examples/mir_demo_server/server_example.cpp](examples/mir_demo_server/server_example.cpp)

Surely it should be enough to add it to `--wayland-extensions`?

File: src/server/frontend_wayland/xdg_output_v1.h [hunk 2]
```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright © 2018 Canonical Ltd.
```
Comment (AlanGriffiths) [inline src/server/frontend_wayland/xdg_output_v1.h]: 2018?

File: src/server/frontend_wayland/xdg_output_v1.h [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Authored by: William Wold <william.wold@canonical.com>
+ */
+
+#ifndef MIR_FRONTEND_XDG_OUTPUT_V1_H
+#define MIR_FRONTEND_XDG_OUTPUT_V1_H
+
+#include "xdg-output-unstable-v1_wrapper.h"
```
Comment (AlanGriffiths) [inline src/server/frontend_wayland/xdg_output_v1.h]: A lot of what is in this file could safely be moved to the .cpp. All that is needed here is a factory function similar to `create_wl_shell()` or `create_data_device_manager()`.

File: src/wayland/symbols.map (hunk 1 above)
Comment (wmww) [reply src/wayland/symbols.map]: Addressed

File: src/server/frontend_wayland/xdg_output_v1.h (hunk 2 above)
Comment (wmww) [reply src/server/frontend_wayland/xdg_output_v1.h]: Fixed

File: src/server/frontend_wayland/xdg_output_v1.h (hunk 3 above)
Comment (wmww) [reply src/server/frontend_wayland/xdg_output_v1.h]: Done

**Comment by wmww** [comment]
I'll write tests for it after https://github.com/MirServer/wlcs/pull/93 lands

**Comment by AlanGriffiths** [comment]
Hmm. rawhide is saying:
```plain
/spread/mir/src/server/console/minimal_console_services.cpp:40:6: error: ambiguating new declaration of â€˜bool drmIsMaster(int)â€™
   40 | bool drmIsMaster(int fd)
      |      ^~~~~~~~~~~
In file included from /spread/mir/src/server/console/minimal_console_services.cpp:32:
/usr/include/xf86drm.h:736:12: note: old declaration â€˜int drmIsMaster(int)â€™
  736 | extern int drmIsMaster(int fd);
```
And F-28 is AWOL

**Comment by wmww** [comment]
bors r=AlanGriffiths

**Pull request merged**